### ISSUE TITLE
r/aws_dax_cluster: Toggling server_side_encryption.enabled should recreate cluster

### ISSUE THREAD

**Issue by ewbankkit** [terraform-providers/terraform-provider-aws#5662]
Changing a DAX cluster's `server_side_encryption.enabled` attribute from `false` to `true` or vice-versa should recreate the cluster as the DAX [`UpdateCluster`](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_dax_UpdateCluster.html) API cannot change the cluster's Encryption value.

```hcl
resource "aws_dax_cluster" "test" {
  cluster_name = "tf-test-001"

  server_side_encryption {
    enabled = true
  }
}
```

=>

```hcl
resource "aws_dax_cluster" "test" {
  cluster_name = "tf-test-001"

  server_side_encryption {
    enabled = false
  }
}
```

```console
$ terraform plan
Refreshing Terraform state in-memory prior to plan...
The refreshed state will be used to calculate this plan, but will not be
persisted to local or remote state storage.

aws_dax_cluster.test: Refreshing state... (ID: tf-test-001)

------------------------------------------------------------------------

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  ~ aws_dax_cluster.test
      server_side_encryption.0.enabled: "false" => "true"


Plan: 0 to add, 1 to change, 0 to destroy.

------------------------------------------------------------------------

Note: You didn't specify an "-out" parameter to save this plan, so Terraform
can't guarantee that exactly these actions will be performed if
"terraform apply" is subsequently run.
```

**Comment by ewbankkit** [comment]
Related:
* https://github.com/terraform-providers/terraform-provider-aws/pull/5508
* https://github.com/terraform-providers/terraform-provider-aws/issues/5500

**Comment by ewbankkit** [comment]
I think this was introduced in https://github.com/terraform-providers/terraform-provider-aws/commit/a55c2bcfa4bdbc8a47f313e0d74aff173bcfc7a6.
We need to add back `ForceNew: true`.

**Comment by bflad** [comment]
Ah sorry about that. 😅

**Comment by bflad** [comment]
This has been released in [version 1.34.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1340-august-30-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
